enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,618 @@
+   * Run the RHEL-HA stack “under to kubelet” (directly on the hardware, not as an OpenShift workload)
+   * Corosync for super fast failure detection, membership calculations, which in turn will trigger Pacemaker to apply Fencing based on Corosync quorum/membership information.
+   * Pacemaker for integrating membership and quorum information, driving fencing, and managing if/when kubelet and etcd can be started
+   * Pacemaker models kubelet and cri-o as a clone (much like a ReplicaSet) and etcd as a “promotable clone” (think a construct designed for leader/follower style services).  Together with fencing and quorum, this ensures that an isolated node that reboots is inert and can do no harm.
+   * Pacemaker is [configured](//TODO mshitrit add link) to manage etcd/cri-o/kubelet, it will start/stop/restart those services using a script or an executable.
+   * Pacemaker does not understand what it is managing, and expects an executable or script that knows how to start/stop/monitor (and optionally promote/demote) the service.
+     1. Likely we would need to create one for etcd
+     2. For kubelet and cri-o we can likely use the existing systemd unit file
+2. Failure Scenarios:
+   * Implement detailed handling procedures for cold boots, network failures, node failures, kubelet failures, and etcd failures using the RHEL-HA stack.
+    [see examples](#failure-handling)
+3. Fencing Methods:
+   * We plan to use Baseboard Management Controller (BMC) as our primary fencing method, the premise of using BMC for fencing is that a node that is powered off, or was previously powered off and configured to be inert until quorum forms, is not in a position to cause corruption or diverging datasets.  Sending power-off (or reboot) commands to the peer’s BMC achieves this goal.
+
+
+### Workflow Description
+
+#### Cluster Creator Role:
+* The Cluster Creator will automatically install the 2NO (by using an installer), installation process will include the following steps:  
+  * Deploys a two-node OpenShift cluster
+  * Configures cluster membership and quorum using Corosync.
+  * Sets up Pacemaker for resource management and fencing.
+
+#### Application Administrator Role:
+* Receives cluster credentials.
+* Deploys applications within the two-node cluster environment.
+
+#### Failure Handling:
+
+1. Cold Boot
+   1. One node (Node1) boots
+   2. Node1 does have “corosync quorum” (C-quorum)  (requires forming a membership with it’s peer)
+   3. Node1 does not start etcd or kubelet, remains inert waiting for Node2
+   4. Peer (Node2) boots
+   5. Corosync membership containing both nodes forms
+   6. Pacemaker “starts” etcd on both nodes
+      * Detail, this could be a “soft”-start which allows us to determine which node has the most recent dataset.
+   7. Pacemaker “promotes” etcd on whichever node has the most recent dataset
+   8. Pacemaker “promotes” etcd on the peer once it has caught up
+   9. Pacemaker starts kubelet on both nodes
```

Comment:
In default OCP, kubelet starts before etcd (1st CRI-O, 2nd kubelet, 3rd etcd).
I think we could keep this same order in this case as well

Reply:
Reflected in the doc. Resolved